src/rust/bitbox02/src/keystore.rs

```diff
@@ -111,3 +113,21 @@ pub fn secp256k1_pubkey_uncompressed(
         false => Err(()),
     }
 }
+
+pub fn encode_xpub_at_keypath(keypath: &[u32], xpub_type: xpub_type_t) -> Result<String, ()> {
+    let mut xpub = [0u8; bitbox02_sys::XPUB_ENCODED_LEN as _];
+    match unsafe {
+        bitbox02_sys::keystore_encode_xpub_at_keypath(
+            keypath.as_ptr(),
+            keypath.len() as _,
+            xpub_type,
+            xpub.as_mut_ptr(),
+            xpub.len() as _,
+        )
+    } {
+        true => Ok(crate::util::str_from_null_terminated(&xpub[..])
+            .unwrap()
```

Comment:
Since the returned value is a `Result`, would be make more sense to propagate an error with `?` instead of panic?

Reply:
What do you think?

In my opinion, panics are appropriate for programming errors that should never happen (independent of user input, IO, hardware environment, etc). In this case, the xpub will always be ascii and null terminated and fit in the `XPUB_ENCODED_LEN` length, and if it doesn't, something is very bad.

Reply:
Yeah sure a valid point but I'm personally more for a gracefull error handling and make users suffer as little as possible, even if it's a programming error, trying to catch as many of those as possible at compile time and in unit tests.